packages/openfeature-js/src/lib/client.ts

```diff
@@ -1,106 +1,237 @@
+    flagId: string,
+    defaultValue: number,
+    context: Context,
+    options?: FlagEvaluationOptions
+  ): Promise<number> {
+    return this.evaluateFlag('number', flagId, defaultValue, context, options);
+  }
+
+  getObjectValue<T extends object>(
+    flagId: string,
     defaultValue: T,
-    context?: Context
+    context: Context,
+    options?: FlagEvaluationOptions
   ): Promise<T> {
-    const span = this.startSpan(`feature flag - ${type}`);
-    try {
-      span.setAttribute(SpanProperties.FEATURE_FLAG_ID, id);
+    return this.evaluateFlag('json', flagId, defaultValue, context, options);
+  }
+
+  private async evaluateFlag<T extends FlagValue>(
+    flagType: FlagType,
+    flagId: string,
+    defaultValue: T,
+    context: Context,
+    options?: FlagEvaluationOptions
+  ): Promise<T> {
+    const provider = this.getProvider();
+    const flagHooks = options?.hooks ?? [];
+    const allHooks: Hook<FlagValue>[] = [
+      ...OpenFeatureAPI.getInstance().hooks,
+      ...this.hooks,
+      ...flagHooks,
+    ];
+    context = context ?? {};
+    let hookContext: HookContext = {
+      flagId,
+      flagType,
+      defaultValue,
+      context,
+      client: this,
+      provider: this.getProvider(),
+    };
+    let valuePromise: Promise<FlagValue>;
 
-      const provider = this.getProvider();
-      let valuePromise: boolean | string | number | object;
-      switch(type) {
+    try {
+      hookContext = this.beforeEvaluation(allHooks, hookContext);
+      switch (flagType) {
+        case 'enabled': {
+          valuePromise = provider.isEnabled(
+            flagId,
+            defaultValue as boolean,
+            context,
+            options
+          );
+          break;
+        }
         case 'boolean': {
-          valuePromise = provider.getBooleanValue(id, defaultValue as boolean, context);
+          valuePromise = provider.getBooleanValue(
+            flagId,
+            defaultValue as boolean,
+            context,
+            options
+          );
           break;
         }
         case 'string': {
-          valuePromise = provider.getStringValue(id, defaultValue as string, context);
+          valuePromise = provider.getStringValue(
+            flagId,
+            defaultValue as string,
+            context,
+            options
+          );
           break;
         }
         case 'number': {
-          valuePromise = provider.getNumberValue(id, defaultValue as number, context);
+          valuePromise = provider.getNumberValue(
+            flagId,
+            defaultValue as number,
+            context,
+            options
+          );
           break;
         }
         case 'json': {
-          valuePromise = provider.getObjectValue(id, defaultValue as object, context);
+          valuePromise = provider.getObjectValue(
+            flagId,
+            defaultValue as object,
+            context,
+            options
+          );
           break;
         }
```

Comment:
Perhaps we should add a default case that throws.

Reply:
I can't see how that would hurt, in the event of some strange typo.